ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/hardware-counters.j2

```diff
@@ -2,7 +2,10 @@
 {% if hardware_counters.features is arista.avd.defined %}
 !
 {%     for feature in hardware_counters.features | arista.avd.natural_sort("name") %}
-{%         set hardware_counters_cli = "hardware counter feature " ~ feature.name %}
+{%         set hardware_counters_cli = "" %}
+{%         if feature.name is arista.avd.defined(fail_action="error", var_name="hardware_counters.features.name") %}
```

Comment:
Maybe - just because it is a breaking change with same key.
```suggestion
{%         if feature.name is arista.avd.defined(fail_action="error", var_name="The data model 'hardware_counters' changed in AVD4.0.0. 'hardware_counters.features.name'") %}
```